src/assemble_workflow/bundle_recorder.py

```diff
@@ -35,7 +36,13 @@ def __get_package_name(self, build):
             build.platform,
             build.architecture,
         ]
-        return "-".join(parts) + (".zip" if build.platform == "windows" else ".tar.gz")
+        if self.distribution == "rpm":
```

Comment:
Add a property called `extension` that uses map and look up the value with that.

```
EXTENSIONS = {
   "rpm": ".rpm",
   "tar.gz": ...
}
```